lightning_examples/finetuning-scheduler/finetuning-scheduler.py

```diff
@@ -0,0 +1,580 @@
+# ---
+# jupyter:
+#   jupytext:
+#     formats: ipynb,py:percent
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.13.1
+#   kernelspec:
+#     display_name: 'Python 3.7.11 64-bit (''pldev_tutorials'': conda)'
+#     language: python
+#     name: python3
+# ---
+
+# %% [markdown]
+# ## Scheduled Finetuning
+#
+# <div style="display:inline" id="a1">
+#
+# The ``FinetuningScheduler`` callback enables multi-phase,
+# scheduled finetuning of foundational models. Gradual unfreezing (i.e. thawing) can help maximize
+# foundational model knowledge retention while allowing (typically upper layers of) the model to
+# optimally adapt to new tasks during transfer learning [1, 2, 3](#f1)
+#
+# </div>
+#
+# ``FinetuningScheduler`` orchestrates the gradual unfreezing
+# of models via a finetuning schedule that is either implicitly generated (the default) or explicitly provided by the user
+# (more computationally efficient). Finetuning phase transitions are driven by
+# ``FTSEarlyStopping`` criteria (a multi-phase
+# extension of ``EarlyStopping``), user-specified epoch transitions or a composition of the two (the default mode).
+# A ``FinetuningScheduler`` training session completes when the
+# final phase of the schedule has its stopping criteria met. See
+# the [early stopping documentation](https://pytorch-lightning.readthedocs.io/en/latest/extensions/generated/pytorch_lightning.callbacks.EarlyStopping.html) for more details on that callback's configuration.
+#
+# <img src="fts_explicit_loss_anim.gif" width="376px" height="272px">
```

Comment:
lets use MarkDown formating

Reply:
Done! The picture is a bit bigger than I was overriding it to be w/ the html but still looks fine I think.

Reply:
well, the advantage of MD formating is that in the publication we will inline it in the notebooks so it becomes standalone with full illustrations...

Reply:
True, that's definitely nice. Would you like me to switch the remaining two illustrations I have to pure MD as well? They look a bit less appealing without the html massaging unfortunately.  I noticed the UvA-DL tutorials had some html img tags I'm assuming for the same reason but if you think it's a worthwhile trade-off I can switch them to pure MD.